server/cli/src/test_connection.rs

```diff
@@ -0,0 +1,499 @@
+        } else {
+            Err(anyhow!("Failed to ping server: {:?}", response))
+        }
+    }
+}
+
+struct DatabaseTest;
+
+#[async_trait]
+impl Test for DatabaseTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded".to_string()))?;
+
+        info!(
+            "Testing database {} on server: {}",
+            config.database.database_name.to_string(),
+            config.database.host.to_string()
+        );
+
+        let connection_manager = get_storage_connection_manager(&config.database);
+        let result = connection_manager.execute("select 1");
+
+        if result.is_ok() {
+            Ok("Successfully connected to database".to_string())
+        } else {
+            Err(anyhow!(
+                "Failed to connect to database: {:?}",
+                result.err().unwrap()
+            ))
+        }
+    }
+}
+
+struct LoginTest;
+
+#[async_trait]
+impl Test for LoginTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded".to_string()))?;
+
+        let username = test_data
+            .args
+            .username
+            .clone()
+            .unwrap_or("admin".to_string());
+        let password = test_data
+            .args
+            .password
+            .clone()
+            .unwrap_or("pass".to_string());
+        // let url = get_url(config)?;
+        let sync_settings = get_sync_settings(config)?;
+
+        info!("Testing login at url: {}", sync_settings.url);
+        info!("    Username: {}", username);
+        info!("    Password: {}", password);
+
+        let client = Client::new();
+        let login_api = LoginApiV4::new(client, Url::parse(&sync_settings.url)?);
+
+        let login_input = LoginInputV4 {
+            username,
+            password,
+            login_type: LoginUserTypeV4::User,
+        };
+
+        let _info = login_api
+            .login(login_input)
+            .await
+            .map_err(|err| anyhow!("Failed to login: {:?}", err))?;
+
+        Ok("Successfully logged in".to_string())
+    }
+}
+
+struct SyncTest;
+
+#[async_trait]
+impl Test for SyncTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded"))?;
+
+        let v5_settings = get_sync_settings(config)?;
+
+        let server_folder = config
+            .server
+            .base_dir
+            .clone()
+            .ok_or(anyhow!("No server base dir in config"))?;
+
+        let hardware_id = test_data.args.server_uuid.clone().unwrap_or(
+            AppDataService::new(server_folder.as_str())
+                .get_hardware_id()
```

Comment:
the hardware_id would only be populated after the server runs, is the use case for this app before or after first server run ? Was also wondering why `--server_uuid` would be a parameter ? How would Admin and User get hardware_id ?